### ISSUE TITLE
Dispatch issue with parametric methods where subtype constraint is a Type

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#12721]
Consider (on 0.4):

``` julia
julia> f{T<:Type{Int}}(::T) = true;

julia> f(Float64)
true
```

I think this should be a MethodError. It seems like `f` is applicable for any DataType, not just `Type{Int}`.

On 0.3, `f(Float64)` is in fact a MethodError, but so is `f(Int)`, which I would expect to work, so while the behavior has changed it doesn't seem like this is a regression.

**Comment by JeffBezanson** [comment]
I think this has the same root cause as #11327.

**Comment by JeffBezanson** [comment]
This is now fixed. (Maybe by #18457 but might have been fixed before that.)

**Comment by tkelman** [comment]
should add a test if there isn't one

**Comment by mauro3** [comment]
This is fixed on 0.4.7 already.
